pkg/transform/transform.go

```diff
@@ -0,0 +1,158 @@
+package transform
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/fusor/cpma/env"
+	"github.com/fusor/cpma/pkg/io"
+	"github.com/fusor/cpma/pkg/transform/oauth"
+	"github.com/fusor/cpma/pkg/transform/secrets"
+	"github.com/sirupsen/logrus"
+)
+
+// OCP4InstallMsg message about using generated manifests
+const OCP4InstallMsg = `To install OCP4 run the installer as follow in order to add CRDs:
+' /openshift-install --dir $INSTALL_DIR create install-config'
+'./openshift-install --dir $INSTALL_DIR create manifests'
+# Copy generated CRD manifest files  to '$INSTALL_DIR/openshift/'
+# Edit them if needed, then run installation:
+'./openshift-install --dir $INSTALL_DIR  create cluster'`
+const MasterConfigFile = "/etc/origin/master/master-config.yaml"
+const NodeConfigFile = "/etc/origin/node/node-config.yaml"
+const RegistriesConfigFile = "/etc/containers/registries.conf"
+
+type Cluster struct {
+	Master Master
+}
+
+type Master struct {
+	OAuth   oauth.OAuthCRD
+	Secrets []secrets.Secret
+}
+
+type Manifest struct {
+	Name string
+	CRD  []byte
+}
+
+type Config struct {
+	MasterConfigFile     string
+	NodeConfigFile       string
+	RegistriesConfigFile string
+	OutputDir            string
+	Hostname             string
+}
+
+type ManifestTransformOutput struct {
+	//Config    Config
+	Manifests []Manifest
+}
+
+type TransformRunner struct {
+	Config string
+}
+
+// GetFile allows to mock file retrieval
+var GetFile = io.GetFile
+
+func Start() {
+	config := Config{}
+	config.OutputDir = env.Config().GetString("OutputDir")
+	config.Hostname = env.Config().GetString("Source")
+	config.MasterConfigFile = MasterConfigFile
+	config.RegistriesConfigFile = RegistriesConfigFile
+	transformRunner := NewTransformRunner(config)
+
+	if err := transformRunner.Transform([]Transform{
+		OAuthTransform{
+			Config: &config,
+		},
+		SDNTransform{
+			Config: &config,
+		},
+		RegistriesTransform{
+			Config: &config,
+		},
+	}); err != nil {
+		logrus.WithError(err).Fatalf("%s", err.Error())
+	}
+}
+
+// DumpManifests creates OCDs files
+func DumpManifests(manifests []Manifest) {
+	for _, manifest := range manifests {
+		maniftestfile := filepath.Join(env.Config().GetString("OutputDir"), "manifests", manifest.Name)
+		os.MkdirAll(path.Dir(maniftestfile), 0755)
+		err := ioutil.WriteFile(maniftestfile, manifest.CRD, 0644)
+		logrus.Printf("CRD:Added: %s", maniftestfile)
+		if err != nil {
+			logrus.Panic(err)
+		}
+	}
+}
+
+func (config *Config) Fetch(path string) []byte {
```

Comment:
[ISSUE] This doesn't feel like it belongs in this file. I've talked about this before, but it would be really nice to have some kind of a type called a `FileManager` who's role it is to deliver requested file contents to the extract methods. In the simplest case, this could just be a vanilla function that has a signature of `GetFile(fileName string)`, but I suspect it will continue to get more complex then that, especially when you consider the fact that you may have to ask for a file from a specific host in the multihost case. Maybe that's just another argument to `GetFile`, and pure functions are easier to test, but this is also a function that, by its nature, has externalities to be considered (other hosts). Just kind of dumping my thoughts, I would not move this for now.

Reply:
On top of that, you could inject a `MockFileManager` for testing extract functions with dummy data, that complies with the `FileManager` interface.